src/server/server_family.cc

```diff
@@ -755,6 +755,18 @@ void ServerFamily::PauseReplication(bool pause) {
   }
 }
 
+std::optional<ReplicaOffsetInfo> ServerFamily::GetReplicaOffsetInfo() {
+  unique_lock lk(replicaof_mu_);
+
+  // Switch to primary mode.
+  if (!ServerState::tlocal()->is_master) {
+    auto repl_ptr = replica_;
+    CHECK(repl_ptr);
```

Comment:
So we just crash if we're neither replica nor master ? 🤔

Reply:
this is the same check taken from ServerFamily::ReplicaOf and in ServerFamily::PauseReplication
if server is replica we expect replica_ to be valid.